pycue/opencue/cuelogging.py

```diff
@@ -0,0 +1,166 @@
+                # being used by another process. Frame execution doesn't need to
+                # be halted for this.
+                if platform.system() == "Windows":
+                    log.warning(err)
+                else:
+                    raise RuntimeError(err)
+            # pylint: disable=consider-using-with
+            self.fd = open(self.filepath, "w+", 1, encoding='utf-8')
+            try:
+                os.chmod(self.filepath, 0o666)
+            # pylint: disable=broad-except
+            except Exception as e:
+                err = "Failed to chmod log file! %s due to %s" % (self.filepath, e)
+                log.warning(err)
+        elif mode == MODE_READ:
+            self.fd = open(self.filepath, "r", encoding='utf-8')
+        else:
+            log.error("Unknown mode for log file!")
+
+    # pylint: disable=arguments-differ
+    def write(self, data, prependTimestamp=False):
+        """Abstract write function that will write to the correct backend"""
+        # Only work if in write mode, otherwise ignore
+        if self.mode == MODE_WRITE:
+            # Convert data to unicode
+            if isinstance(data, bytes):
+                data = data.decode('utf-8', errors='ignore')
+            if prependTimestamp is True:
+                lines = data.splitlines()
+                curr_line_timestamp = datetime.datetime.now().strftime("%H:%M:%S")
+                for line in lines:
+                    print("[%s] %s" % (curr_line_timestamp, line), file=self)
+            else:
+                self.fd.write(data)
+
+    def writelines(self, __lines):
+        """Provides support for writing mutliple lines at a time"""
+        for line in __lines:
+            self.write(line)
+
+    def close(self):
+        """Closes the file if the backend is file based"""
+        self.fd.close()
+
+    def waitForFile(self, maxTries=5):
+        """Waits for the file to exist before continuing when using a file backend"""
+        # Waits for a file to exist
+        tries = 0
+        while tries < maxTries:
+            if os.path.exists(self.filepath):
+                return
+            tries += 1
+            time.sleep(0.5 * tries)
+        raise IOError("Failed to create %s" % self.filepath)
+
+    def size(self):
+        """Return the size of the file"""
+        return int(os.stat(self.filepath).st_size)
+
+    def getMtime(self):
+        """Return modification time of the file"""
+        return os.path.getmtime(self.filepath)
+
+    def exists(self):
+        """Check if the file exists"""
+        return os.path.exists(self.filepath)
+
+    def read(self):
+        """Read the data from the backend"""
+        # Only allow reading when in read mode
+
+        content = None
+        if self.mode == MODE_READ:
+            if self.exists() is True:
+                with open(self.filepath, "r", encoding='utf-8') as fp:
+                    content = fp.read()
+
+        return content
```

Comment:
This function can be hard to interact with due to concentrating multiple exits on a single return type (None). 
 - If trying to read on the wrong mode 
 - If the file doesn't exist
I suggest raising an exception to be handled accordingly if necessary, or at least documenting what's expected as return.

Reply:
Have added a IOError if the file doesn't exist